### PR TITLE
Async diagnostic node

### PR DESCRIPTION
**Pull request by jacob314**

### CODE REVIEW

**Comment by jacob314** [comment]
PQ reviewed the similar CL in the intellij code.
This will simplify things in the logging view./ It isn't really needed on the web client but I want to keep the clients consistent.

File: lib/inspector/diagnostics_node.dart [hunk 1]
```diff
@@ -405,8 +404,8 @@ class RemoteDiagnosticsNode extends DiagnosticableTree {
           _valueProperties = Future.value(null);
           return _valueProperties;
       }
-      _valueProperties =
-          inspectorService.getDartObjectProperties(valueRef, propertyNames);
+      _valueProperties = (await inspectorService)
+          ?.getDartObjectProperties(valueRef, propertyNames);
```
Comment (pq) [inline lib/inspector/diagnostics_node.dart]: And here we go.... SO MUCH NICER than the Java version.

File: lib/inspector/inspector_controller.dart [hunk 2]
```diff
@@ -244,9 +246,7 @@ class InspectorController implements InspectorServiceClient {
     subtreeRoot = null;
 
     inspectorTree.root = inspectorTree.createNode();
-    if (details != null) {
-      details.shutdownTree(isolateStopped);
-    }
+    details?.shutdownTree(isolateStopped);
```
Comment (pq) [inline lib/inspector/inspector_controller.dart]: 💎

File: lib/inspector/inspector_controller.dart [hunk 3]
```diff
@@ -774,8 +754,7 @@ class InspectorController implements InspectorServiceClient {
       // We can't rely on the details tree to update the selection on the server in this case.
       final selection =
           detailsSelection != null ? detailsSelection : diagnostic;
```
Comment (pq) [inline lib/inspector/inspector_controller.dart]: Maybe:

```dart
final selection = detailsSelection ?? diagnostic;
```

?

Comment (jacob314) [reply lib/inspector/inspector_controller.dart]: done

**Comment by pq** [comment]
Upgrading my approval from tentative to emphatic after a quick chat with @jacob314.

🚀

**Pull request merged**